httpapi/httpapi.go

```diff
@@ -104,3 +108,14 @@ func (h HTTPAPI) deleteConnection(w http.ResponseWriter, r *http.Request, params
 type ConnectionsResponse struct {
 	Connections []*connection.Connection `json:"connections"`
 }
+
+func extractMetadataFilters(query url.Values) (filters []metadata.Filter) {
+	prefix := "metadata."
+
+	for key, value := range query {
+		if strings.HasPrefix(key, prefix) {
+			filters = append(filters, metadata.Filter{Key: strings.TrimPrefix(key, prefix), Value: value[0]})
+		}
+	}
+	return
```

Comment:
Oof, naked returns?! I would just amend to `return filters` even though it does the same thing.